src/test/scala/beam/agentsim/infrastructure/taz/H3TAZSpec.scala

```diff
@@ -0,0 +1,78 @@
+import beam.sim.{BeamHelper, BeamServices}
+import beam.sim.config.{BeamConfig, MatSimBeamConfigBuilder}
+import beam.sim.population.DefaultPopulationAdjustment
+import beam.utils.FileUtils
+import beam.utils.TestConfigUtils.testConfig
+import com.typesafe.config.{Config, ConfigFactory}
+import org.matsim.api.core.v01.events.Event
+import org.matsim.api.core.v01.population.Activity
+import org.matsim.core.controler.AbstractModule
+import org.matsim.core.events.handler.BasicEventHandler
+import org.matsim.core.scenario.{MutableScenario, ScenarioUtils}
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.collection.JavaConverters._
+
+class H3TAZSpec extends FlatSpec with Matchers with BeamHelper {
+
+  "test demand inferred H3 Index set" must "H3 Indexes" in {
+    val config = ConfigFactory
+      .parseString("""
+                     |beam.agentsim.h3taz = {
+                     |  lowerBoundResolution = 6
+                     |  upperBoundResolution = 9
+                     |}
+                     |beam.physsim.skipPhysSim = true
+                     |beam.agentsim.lastIteration = 0
+        """.stripMargin)
+      .withFallback(testConfig("test/input/sf-light/sf-light-1k.conf"))
+      .resolve()
+    runGetDemandInferredH3IndexSet(config)
+  }
+
+  def runGetDemandInferredH3IndexSet(config: Config): Unit = {
+    val configBuilder = new MatSimBeamConfigBuilder(config)
+    val matsimConfig = configBuilder.buildMatSimConf()
+    val beamConfig = BeamConfig(config)
+    val beamScenario = loadScenario(beamConfig)
+    FileUtils.setConfigOutputFile(beamConfig, matsimConfig)
+    val scenario = ScenarioUtils.loadScenario(matsimConfig).asInstanceOf[MutableScenario]
+    scenario.setNetwork(beamScenario.network)
+    val injector = org.matsim.core.controler.Injector.createInjector(
+      scenario.getConfig,
+      new AbstractModule() {
+        override def install(): Unit = {
+          install(module(config, beamConfig, scenario, beamScenario))
+        }
+      }
+    )
+    val services = injector.getInstance(classOf[BeamServices])
+    DefaultPopulationAdjustment(services).update(scenario)
+    val demandPoints = scenario.getPopulation.getPersons.asScala
+      .flatMap(_._2.getPlans.get(0).getPlanElements.asScala.filter(_.isInstanceOf[Activity]))
+      .map(_.asInstanceOf[Activity].getCoord)
+      .toArray
+    val maxDemandPoints = 100
+    val lowestResolution = 6
+    val highestResolution = 9
+    val indexes =
+      H3TAZ.getDataPointsInferredH3IndexSet(demandPoints, maxDemandPoints, lowestResolution, highestResolution)
+    assert(indexes.nonEmpty, "Set of H3 indexes should not be empty.")
+    assert(indexes.map(_._2.length).sum == demandPoints.length, "Not all coordinates were matched to an H3 Index")
+    assert(
+      indexes.map(_._2.length).max <= maxDemandPoints,
+      "number of data points should be lower than or equal to 100"
+    )
+    assert(
```

Comment:
how about we test that the output indexes are in the correct range (we could omit testing the config here):

// Check that all indexes are within constrain
indexes.foreach { case (h3Index, coords) =>
  val resolution = H3TAZ.H3.h3GetResolution(h3Index)
  require(resolution >= lowestResolution && resolution <= highestResolution, s"Expected to have resolution in [$lowestResolution, $highestResolution], but got $resolution")
}

for this to work you might have to make H3TAZ.H3 public or provide a getter